src/systems/systems/src/integrated_modular_avionics/avionics_full_duplex_switch.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    shared::{power_supply_relay::PowerSupplyRelay, ElectricalBusType, ElectricalBuses},
+    simulation::{
+        InitContext, Read, SimulationElement, SimulationElementVisitor, SimulatorReader,
+        SimulatorWriter, VariableIdentifier, Write,
+    },
+};
+use std::option::Option;
+
+pub struct AvionicsFullDuplexSwitch {
+    single_power_supply: Option<ElectricalBusType>,
+    power_supply_relay: Option<PowerSupplyRelay>,
+    last_is_powered: bool,
+    is_powered: bool,
+    failure_indication_id: VariableIdentifier,
+    last_failure_indication: bool,
+    failure_indication: bool,
+    available_id: VariableIdentifier,
```

Comment:
see rust guidelines.md and readme.md ideally systems should not be aware they are running in a simulator and simvar state should be passed and vars not being read here i.e. the system is unaware that it is running in a simulator

Reply:
Resolved in development channel on discord